### PR TITLE
>=media-video/ffmpeg-3: Reintroduce libressl USE

### PR DESCRIPTION
**Pull request by lluixhi**
Add patch for ffmpeg-3.2.

I'm not sure why the use flag was dropped.

### CODE REVIEW

File: media-video/ffmpeg/ffmpeg-3.1.5.ebuild [hunk 1]
```diff
@@ -214,7 +214,10 @@ RDEPEND="
 	modplug? ( >=media-libs/libmodplug-0.8.8.4-r1[${MULTILIB_USEDEP}] )
 	openal? ( >=media-libs/openal-1.15.1[${MULTILIB_USEDEP}] )
 	opengl? ( >=virtual/opengl-7.0-r1[${MULTILIB_USEDEP}] )
-	openssl? ( >=dev-libs/openssl-1.0.1h-r2:0[${MULTILIB_USEDEP}] )
+	openssl? (
+		!libressl? ( >=dev-libs/openssl-1.0.1h-r2:0[${MULTILIB_USEDEP}] )
+		libressl? ( dev-libs/libressl[${MULTILIB_USEDEP}] )
```
Comment (SoapGentoo) [inline media-video/ffmpeg/ffmpeg-3.1.5.ebuild]: both libressl and openssl needs `:0=`

File: media-video/ffmpeg/ffmpeg-3.2.ebuild [hunk 2]
```diff
@@ -213,7 +213,10 @@ RDEPEND="
 	modplug? ( >=media-libs/libmodplug-0.8.8.4-r1[${MULTILIB_USEDEP}] )
 	openal? ( >=media-libs/openal-1.15.1[${MULTILIB_USEDEP}] )
 	opengl? ( >=virtual/opengl-7.0-r1[${MULTILIB_USEDEP}] )
-	openssl? ( >=dev-libs/openssl-1.0.1h-r2:0[${MULTILIB_USEDEP}] )
+	openssl? (
+		!libressl? ( >=dev-libs/openssl-1.0.1h-r2:0[${MULTILIB_USEDEP}] )
+		libressl? ( dev-libs/libressl[${MULTILIB_USEDEP}] )
```
Comment (SoapGentoo) [inline media-video/ffmpeg/ffmpeg-3.2.ebuild]: same here

File: media-video/ffmpeg/ffmpeg-3.1.5.ebuild (hunk 1 above)
Comment (lluixhi) [inline media-video/ffmpeg/ffmpeg-3.1.5.ebuild]: Sure. I didn't touch it because this was the convention in previous versions.

**Comment by lluixhi** [comment]
@SoapGentoo 

Found the bug that was the reason for dropping libressl support:
https://bugs.gentoo.org/show_bug.cgi?id=569818

**Comment by doughdemon** [comment]
So shouldn't that bug then be fixed first? (Introduce an ssl use flag.)

**Comment by SoapGentoo** [comment]
This is a general Gentoo problem: the exact semantics between `ssl`, `gnutls`, `openssl` and `libressl` aren't well defined (Beyond the _libressl_<->_openssl_ semantics). I think this should be clarified.

**Comment by lluixhi** [comment]
@doughdemon 
The reason the ssl USE flag didn't replace the openssl USE was because the ssl USE flag means that an SSL/TLS implementation is going to be used.

Unfortunately, ffmpeg only supports openssl for TLS.. and it enables a license exception.

@SoapGentoo 
Where's a good place to discuss this?

**Comment by SoapGentoo** [comment]
@lluixhi in general the take is that USE="ssl" enable SSL support, USE="openssl" enables OpenSSL or LibreSSL, USE="yassl" for yassl and so on.

**Comment by doughdemon** [comment]
Aric Belsito wrote:
> @doughdemon 
> The reason the ssl USE flag didn't replace the openssl USE was because the ssl USE flag means that an SSL/TLS implementation is going to be used.
> 
> Unfortunately, ffmpeg only supports openssl for TLS.. and it enables a license exception.

Sorry I'm a bit confused: Isn't your patch fixing (at least
compilation) for libressl so that ssl could be enabled also when
libressl is used? Doesn't the license exception also apply to
libressl?

**Comment by lluixhi** [comment]
Yes, the license exception also applies to LibreSSL -- but the maintainer understood the ssl USE flag to suggest the possibility of gnutls.

**Comment by aballier** [comment]
the change does not fix the above mentionned bug, so it wont be merged as such

Main issue I still have is: Do we have some coordination on the libressl side these days or do we continue to push global useflags that have different meanings for different packages ?

**Comment by blueness** [comment]
@aballier yes we do have a libressl point-of-contact.  me and @SoapGentoo.  We're committed to getting libressl into the tree.

@SoapGentoo   I'm all about cleaning up the USE flag semantics but I'd like that discussion to happen on our gentoo bugzilla (or at least a summary there) since its our infra and we want to keep a record of what decisions we take.

**Comment by lluixhi** [comment]
Opened a bug here for discussion:
https://bugs.gentoo.org/show_bug.cgi?id=611564

It'd be nice if we could resolve this.

**Comment by mgorny** [comment]
I'm sorry but I don't think this is going to get merged until developers can sort out the problems between themselves. Feel free to reopen if anything comes out of the bug.

**Pull request closed without merging**